packages/entities/entities-gateway-services/src/components/GatewayServiceForm.vue

```diff
@@ -587,34 +672,248 @@ const changeCheckedGroup = () => {
+  const portNum = typeof port === 'string' ? parseInt(port, 10) : port
+
+  // Check if port is a valid number
+  if (isNaN(portNum)) {
+    return 'Port must be a number'
+  }
+
+  // Check if port is in valid range (0-65535)
+  if (portNum < 0 || portNum > 65535) {
+    return 'Port must be between 0 and 65535'
+  }
+
+  // Valid port
+  return ''
+}
+
+const validatePath = (path: string | null | undefined): string => {
+  // Path is not required, so return empty string if not provided
+  if (!path || path === '') {
+    return ''
+  }
+
+  // If path exists, it should start with '/'
+  if (!path.startsWith('/')) {
+    return 'Path must begin with /'
+  }
+
+  // This regex matches any character that is NOT allowed by RFC 3986
+  const invalidCharsRegex = /[^A-Za-z0-9\-._~:/?#[\]@!$&'()*+,;=%]/
+
+  if (invalidCharsRegex.test(path)) {
+    return 'Path should not include characters outside of the reserved list of RFC 3986'
+  }
+
+  // Valid path according to RFC 3986
+  return ''
+}
+
+const handleValidateFullUrl = useDebounceFn((): void => {
+  // reset the errors
+  resetFormFieldErrors()
+
+  if (form.fields.url.length) {
     try {
+      // try constructing a url
       const parsedUrl = new URL(form.fields.url)
-      // if the URL is parsed successfully, clear error message
-      form.errorMessage = ''
+      // url is valid
 
-      // remove the trailing colon appended to parsedUrl.protocol
+      // validate protocol
+      const protocolError = validateProtocol(parsedUrl.protocol)
+      if (protocolError) throw new Error(protocolError)
       form.fields.protocol = parsedUrl.protocol.slice(0, -1)
+
+      // validate hostname
+      const hostError = validateHost(parsedUrl.hostname)
+      if (hostError) throw new Error(hostError)
       form.fields.host = parsedUrl.hostname
+
+      // validate path
+      const pathError = validatePath(parsedUrl.pathname)
+      if (pathError) throw new Error(pathError)
       form.fields.path = parsedUrl.pathname
 
+      // validate port
+      const portError = validatePort(parsedUrl.port)
+      if (portError) throw new Error(portError)
       // extract port value from url and convert to number
       const portValue = Number(parsedUrl.port)
-
       hasExtractPortValue.value = !!portValue
       form.fields.port = portValue || getPort.getPortFromProtocol(form.fields.protocol)
 
-      form.errorMessage = ''
       emit('url-valid:success')
     } catch (error: any) {
-      form.errorMessage = t('errors.urlErrorMessage')
-      emit('url-valid:error', getMessageFromError(error))
+      emit('url-valid:error', error.message || 'URL validation failed')
+      form.formFieldErrors.url = error.message || 'URL validation failed'
     }
-  } else {
+  }
+}, 300)
+
+const handleValidateAdvancedFields = useDebounceFn((fieldId?: keyof FormFieldErrors) => {
+  // reset the errors
+  resetFormFieldErrors(fieldId ?? undefined)
+
+}, 300)
+
+const getFieldErrorById = computed(() => {
```

Comment:
question: does this work? it seems odd to make a computed return a function. I'm not really sure if the reactivity will survive this and I'm also not sure Vue is smart enough to realize you want to react to `form.formFieldErrors` when your usage is nested in the `return`. Why aren't you just using a regular function and calling it when you need to.